### ISSUE TITLE
Init system detection fails on Ubuntu 16.04

### ISSUE THREAD

**Issue by vladvasiliu** [saltstack-formulas/docker-formula#121]
Hi,

I'm having the following problem on Ubuntu 16.04:

```
          ID: docker-container-startup-config-exporter
    Function: file.managed
      Result: False
     Comment: Specified file docker-container-startup-config-exporter is not an absolute path
```

Investigating the logs, it turns out the command to identify the init system fails for some reason.

Minion log :

```
2017-09-13 16:25:36,232 [salt.loaded.int.module.cmdmod][INFO    ][7720] Executing command 'ps -p1 | grep -q systemd && echo systemd || echo upstart' in directory '/root'
2017-09-13 16:25:36,240 [salt.loaded.int.module.cmdmod][ERROR   ][7720] Command 'ps -p1 | grep -q systemd && echo systemd || echo upstart' failed with return code: 1
2017-09-13 16:25:36,241 [salt.loaded.int.module.cmdmod][ERROR   ][7720] output: error: garbage option

Usage:
 ps [options]

 Try 'ps --help <simple|list|output|threads|misc|all>'
  or 'ps --help <s|l|o|t|m|a>'
 for additional help text.

For more details see ps(1).
```

However, on the salt master:

```
salt 'elasticsearch-a.logs' cmd.run "ps -p1 | grep -q systemd && echo systemd || echo upstart"
elasticsearch-a.logs:
    systemd
```

And directly on the server:

```
» ps -p1 | grep -q systemd && echo systemd || echo upstart                                                  130 ↵
systemd
```

Replacing line 14 of ````containers.sls```` to use grains instead of a command, it works:

````jinja
{# {%- set init_system = salt["cmd.run"]("ps -p 1 | grep -q systemd && echo systemd || echo upstart") %} #}
{%- set init_system = grains['init'] %}
````

Versions:

Master:

```
» salt -V
Salt Version:
           Salt: 2017.7.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.9.5
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.6
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Jan 19 2017, 14:48:08)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: 4.2.1

System Versions:
           dist: Ubuntu 17.04 zesty
         locale: UTF-8
        machine: x86_64
        release: 4.10.0-33-generic
         system: Linux
        version: Ubuntu 17.04 zesty
```

Minion:

```
» salt-minion -V
Salt Version:
           Salt: 2017.7.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-93-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by blbradley** [comment]
Thanks for the detailed report, @vladvasiliu!

Can you submit a pull request? This issue was also talked about in #119.

**Comment by vladvasiliu** [comment]
PR submitted.

**Comment by blbradley** [comment]
Closed by #122
